src/operator/numpy/random/np_location_scale_op.h

```diff
@@ -0,0 +1,449 @@
+                                  const Shape<ndim> &oshape, IType *array,
+                                  float scalar, float *noise, OType *out) {
+    Shape<ndim> coord = unravel(i, oshape);
+    auto idx = static_cast<index_t>(dot(coord, stride));
+    IType loc_value;
+    IType scale_value;
+    if (scalar_pos == 0) {
+      loc_value = scalar;
+      scale_value = array[idx];
+    } else {
+      loc_value = array[idx];
+      scale_value = scalar;
+    }
+    noise[i] = -log(-log(noise[i]));
+    out[i] = loc_value + noise[i] * scale_value;
+  }
+};
+
+struct logistic_kernel {
+  template <int ndim, typename IType, typename OType>
+  MSHADOW_XINLINE static void Map(index_t i, const Shape<ndim> &lstride,
+                                  const Shape<ndim> &hstride,
+                                  const Shape<ndim> &oshape, IType *loc,
+                                  IType *scale, float *noise, OType *out) {
+    Shape<ndim> coord = unravel(i, oshape);
+    auto lidx = static_cast<index_t>(dot(coord, lstride));
+    auto hidx = static_cast<index_t>(dot(coord, hstride));
+    IType loc_value = loc[lidx];
+    IType scale_value = scale[hidx];
+    noise[i] = log(noise[i]) - log(1 - noise[i]);
+    out[i] = loc_value + noise[i] * scale_value;
+  }
+};
+
+struct gumbel_kernel {
+  template <int ndim, typename IType, typename OType>
+  MSHADOW_XINLINE static void Map(index_t i, const Shape<ndim> &lstride,
+                                  const Shape<ndim> &hstride,
+                                  const Shape<ndim> &oshape, IType *loc,
+                                  IType *scale, float *noise, OType *out) {
+    Shape<ndim> coord = unravel(i, oshape);
+    auto lidx = static_cast<index_t>(dot(coord, lstride));
+    auto hidx = static_cast<index_t>(dot(coord, hstride));
+    IType loc_value = loc[lidx];
+    IType scale_value = scale[hidx];
+    noise[i] = -log(-log(noise[i]));
+    out[i] = loc_value + noise[i] * scale_value;
+  }
+};
+
+}  // namespace mxnet_op
+
+template <typename xpu, typename two_scalar_kernel, typename scalar_tensor_kernel,
+          typename two_tensor_kernel>
+void NumpyLocationScaleForward(const nnvm::NodeAttrs &attrs,
+                               const OpContext &ctx,
+                               const std::vector<TBlob> &inputs,
+                               const std::vector<OpReqType> &req,
+                               const std::vector<TBlob> &outputs) {
+  using namespace mshadow;
+  using namespace mxnet_op;
+  const NumpyLocationScaleParam &param = nnvm::get<NumpyLocationScaleParam>(attrs.parsed);
+  Stream<xpu> *s = ctx.get_stream<xpu>();
+  // Generate base random number.
+  Random<xpu, float> *prnd = ctx.requested[0].get_random<xpu, float>(s);
+  Tensor<xpu, 1, float> workspace =
+    ctx.requested[1].get_space_typed<xpu, 1, float>(Shape1(1), s);
+  Tensor<xpu, 1, float> location_tensor = outputs[1].FlatTo1D<xpu, float>(s);
```

Comment:
Could you change the name to `uniform_tensor` or `noise_tensor`? The name `location_tensor` looks confusing to me, cuz I believe the location is stored in inputs[0].